src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenConstructorInitTests.cs

```diff
@@ -607,5 +737,488 @@ public void BadExpressionConstructor()
                 //     dynamic d = F() * 2;
                 Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "F()").WithArguments("Microsoft.CSharp.RuntimeBinder.CSharpArgumentInfo", "Create").WithLocation(4, 17));
         }
+
+        [WorkItem(42985, "https://github.com/dotnet/roslyn/issues/42985")]
+        [Fact]
+        public void SkipSynthesizedStaticConstructor_01()
+        {
+            string source = @"
+#nullable enable
+class C
+{
+    static int i = 0;
+    static bool b = false;
+}";
```

Comment:
Consider including all primitive types:
```
static sbyte s1 = 0;
static byte s2 = 0;
static short s3 = 0;
static ushort s4 = 0;
static int s5 = 0;
static uint s6 = 0;
static long s7 = 0;
static ulong s8 = 0;
static nint s9 = 0;
static nuint s10 = 0;
static bool s11 = false;
static char s12 = '\0';
static float s13 = 0f;
static double s14 = 0d;
static System.IntPtr s15 = (System.IntPtr)0;
static System.UIntPtr s16 = (System.UIntPtr)0;
static void* s17 = (void*)0;
```
And consider tests of the same types with `default` and `default(T)`.

Reply:
It seems the explicit pointer conversions require things to be emitted today, so i will factor those into their own test, but otherwise I am going to handle this by expanding `CodeGenTests.Regress530049`.